client/signup/navigation-link/test/index.jsx

```diff
@@ -102,44 +108,58 @@ describe( 'NavigationLink', () => {
 		getFilteredSteps.mockReturnValue( [
 			{ stepName: 'test:step1', stepSectionName: 'test:section1', wasSkipped: false },
 		] );
-		wrapper.setProps( { stepName: 'test:step1' } ); // set the first step
+
+		rerender(
+			<NavigationLink { ...props } userLoggedIn={ false } direction="back" stepName="test:step1" />
+		);
 		expect( getStepUrl ).toHaveBeenCalled();
 		expect( getStepUrl ).toHaveBeenCalledWith( 'test:flow', null, '', 'en', undefined );
 
 		// The href should be backUrl when exist.
-		wrapper.setProps( { backUrl: 'test:back-url' } );
-		expect( wrapper.props().href ).toEqual( 'test:back-url' );
+		rerender(
+			<NavigationLink
+				{ ...props }
+				userLoggedIn={ false }
+				direction="back"
+				backUrl="test:back-url"
+			/>
+		);
+		expect( container.firstChild ).toHaveAttribute( 'href', 'test:back-url' );
 	} );
 
-	test( 'should call goToNextStep() only when the direction is forward and clicked', () => {
-		const wrapper = shallow( <NavigationLink { ...props } direction="forward" /> );
+	test( 'should call goToNextStep() only when the direction is forward and clicked', async () => {
+		const user = userEvent.setup();
+		const { container } = render( <NavigationLink { ...props } direction="forward" /> );
 
 		expect( props.goToNextStep ).not.toHaveBeenCalled();
-		wrapper.simulate( 'click' );
+		await user.click( container.firstChild );
```

Comment:
Can we query by the button role here again? Same for the other tests that do this.